lib/bundle/brew_installer.rb

```diff
@@ -1,18 +1,65 @@
 module Bundle
   class BrewInstaller
     def self.install(name, options = {})
+      new(name, options).install_or_upgrade
+    end
+
+    def initialize(name, options = {})
       if Bundle.brew_installed?
-        command = [name]
-        unless options[:args].nil?
-          options[:args].each do |arg|
-            command << "--#{arg}"
-          end
-        end
-
-        Bundle.system "brew", "install", *command
+        @name = name
```

Comment:
Since the argument can be full name (e.g. `homebrew/tap/foo`) or short name (e.g. `bar`). We can do this:

```ruby
@full_name = name
@name = name.split("/").last
```